src/Microsoft.AspNetCore.BuildTools.ApiCheck/ApiListingComparer.cs

```diff
@@ -13,128 +13,108 @@ public class ApiListingComparer
+                        breakingChanges.Add(new BreakingChange(type.Id, /*memberId:*/ null, ChangeKind.Removal));
+                        continue;
                     }
 
-                }
-
-                if (newType != null)
-                {
-                    CompareMembers(type, newType, exclusions, breakingChanges);
+                    CompareMembers(type, newType, breakingChanges);
                 }
             }
 
-            return new ApiComparisonResult(breakingChanges, exclusions);
+            return breakingChanges;
         }
 
-        private void CompareMembers(TypeDescriptor type, TypeDescriptor newType, List<ApiChangeExclusion> exclusions, List<BreakingChange> breakingChanges)
+        private void CompareMembers(TypeDescriptor type, TypeDescriptor newType, List<BreakingChange> breakingChanges)
         {
-            var removedOrChanged = 0;
+            var newMembers = newType.Members.ToList();
+
             foreach (var member in type.Members)
             {
-                var newMember = newType.FindMember(member.Id);
-                var isAcceptable = IsAcceptableMemberChange(_newApiListing, newType, member);
-                if (isAcceptable)
+                if (IsAcceptableMemberChange(newType, member, out var newMember))
                 {
-                    if (newMember == null)
-                    {
-                        removedOrChanged++;
-                    }
-
-                    continue;
+                    newMembers.Remove(newMember);
                 }
-
-                if (newMember == null)
+                else
                 {
-                    removedOrChanged++;
-                    var memberChange = FilterExclusions(type, member, exclusions);
-                    if (memberChange != null)
-                    {
-                        breakingChanges.Add(memberChange);
-                    }
+                    breakingChanges.Add(new BreakingChange(type.Id, member.Id, ChangeKind.Removal));
                 }
             }
 
-            if (type.Kind == TypeKind.Interface && type.Members.Count - removedOrChanged < newType.Members.Count)
+            if (type.Kind == TypeKind.Interface && newMembers.Count > 0)
             {
-                var members = newType.Members.ToList();
-                foreach (var member in newType.Members)
-                {
-                    var change = FilterExclusions(type, null, exclusions);
-                    if (change == null)
-                    {
-                        members.Remove(member);
-                    }
-                }
-
-                if (type.Members.Count - removedOrChanged < members.Count)
-                {
-                    breakingChanges.Add(new BreakingChange(type, "New members were added to the following interface"));
-                }
+                breakingChanges.AddRange(newMembers.Select(member => new BreakingChange(newType.Id, member.Id, ChangeKind.Addition)));
```

Comment:
This part finds actually newly added members on interface (or renamed ones because they are remove/add)